src/argilla/server/app.py

```diff
@@ -12,17 +12,262 @@
+    api.exception_handler(EntityAlreadyExistsError)(APIErrorHandler.common_exception_handler)
+    api.exception_handler(ClosedDatasetError)(APIErrorHandler.common_exception_handler)
+    api.exception_handler(ValidationError)(APIErrorHandler.common_exception_handler)
+    api.exception_handler(AssertionError)(APIErrorHandler.common_exception_handler)
+    api.exception_handler(WrongTaskError)(APIErrorHandler.common_exception_handler)
+    api.exception_handler(MissingInputParamError)(APIErrorHandler.common_exception_handler)
+    api.exception_handler(RequestValidationError)(APIErrorHandler.common_exception_handler)
+    api.exception_handler(InvalidTextSearchError)(APIErrorHandler.common_exception_handler)
+    api.exception_handler(BadRequestError)(APIErrorHandler.common_exception_handler)
+
+
+def configure_api_router(app: FastAPI):
+    """Configures and set the api router to app"""
+    app.include_router(api_router, prefix="/api")
+
+
+def configure_app_statics(app: FastAPI):
+    """Configure static folder for app"""
+
+    parent_path = Path(__file__).parent.absolute()
+    statics_folder = Path(os.path.join(parent_path, "static"))
+    if not (statics_folder.exists() and statics_folder.is_dir()):
+        return
+
+    def _create_statics_folder(path_from):
+        """
+        Application statics will be created with a parameterized baseUrl variable.
+
+        This function will replace the variable by the real runtime value found in settings.base_url
+
+        This allow us to deploy the argilla server under a custom base url, even when webapp does not
+        support it.
+
+        """
+        BASE_URL_VAR_NAME = "@@baseUrl@@"
+        temp_dir = tempfile.mkdtemp()
+        new_folder = shutil.copytree(path_from, temp_dir + "/statics")
+        base_url = helpers.remove_suffix(settings.base_url, suffix="/")
+        for extension in ["*.js", "*.html"]:
+            for file in glob.glob(
+                f"{new_folder}/**/{extension}",
+                recursive=True,
+            ):
+                helpers.replace_string_in_file(
+                    file,
+                    string=BASE_URL_VAR_NAME,
+                    replace_by=base_url,
+                )
+
+        return new_folder
+
+    temp_statics = _create_statics_folder(statics_folder)
+
+    app.mount(
+        "/",
+        RewriteStaticFiles(directory=temp_statics, html=True, check_dir=False),
+        name="static",
+    )
+
+
+def configure_storage(app: FastAPI):
+    def _on_backoff(event):
+        _LOGGER.warning(
+            f"Connection to {settings.obfuscated_elasticsearch()} is not ready. "
+            f"Tried {event['tries']} times. Retrying..."
         )
 
-    app = fallback_app
+    @backoff.on_exception(
+        lambda: backoff.constant(interval=15),
+        ConfigError,
+        max_time=60,
+        on_backoff=_on_backoff,
+    )
+    def _setup_elasticsearch():
```

Comment:
Maybe we  should have a similar a similar function for Redis that do a `ping` to Redis (not here in your PR, I mean in my PR #4427) so we now early if Redis is not available, showing an error message.